exo/networking/manual/manual_discovery.py

```diff
@@ -67,3 +61,18 @@ async def task_find_peers_from_config(self):
 
       if DEBUG_DISCOVERY >= 2: print(f"Current known peers: {[peer.id() for peer in self.known_peers.values()]}")
 
+  def _get_peers(self):
+    try:
+        topology = NetworkTopology.from_path(self.network_config_path)
+
+        if self.node_id not in topology.peers:
+          raise ValueError(f"Node ID {self.node_id} not found in network config file {self.network_config_path}. Please run with `node_id` set to one of the keys in the config file: {[k for k, _ in topology.peers]}")
+
+        peers_in_network: Dict[str, PeerConfig] = topology.peers
+        peers_in_network.pop(self.node_id)
+    except Exception as e:
+        if DEBUG_DISCOVERY >= 2: print(f"Error when loading network config file from {self.network_config_path}. Please update the config file in order to successfully discover peers. Exception: {e}")
+        peers_in_network = {}
```

Comment:
The only gotcha I found with handling dynamic updates to the config file is if someone is manually changing the json (instead of pasting in a fully new & correct json blob) it could be loaded/read in a WIP/incorrect state.

So my solution to that is to just swallow any exception we get when reading from the config and return an empty dict (so there will be nothing for `task_find_peers_from_config` to iterate over and load until the config file is set correctly)